modules/tasks/tasks/src/main/java/org/motechproject/tasks/ex/ProviderNotFoundException.java

```diff
@@ -0,0 +1,11 @@
+package org.motechproject.tasks.ex;
+
+/**
+ * Signals that provider with a given id was not found.
+ */
+public class ProviderNotFoundException extends IllegalArgumentException {
+
+    public ProviderNotFoundException(long providerId) {
+        super("Unable to migrate task. Data provider with id " + providerId + " not found");
```

Comment:
Including the task name for which the migration has failed would be helpful as well.